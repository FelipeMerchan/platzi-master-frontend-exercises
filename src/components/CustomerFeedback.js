import React from 'react'

import './styles/CustomerFeedback.css'

function CustomerFeedback() {
  return (
    <article className="CustomerFeedback">
      <h2 className="CustomerFeedback__title">feedback
        <br /> <span>de clientes</span>
      </h2>
      <div className="feedback">
        <p>
          <strong>Sra. Lorem Ipsum</strong>
        </p>
        <p>
          <span>"Lorem ipsum dolot sit amet, muy buen trabajo!"</span>
        </p>
      </div>
      <div className="feedback">
        <p>
          <strong>Sra. Lorem Ipsum</strong>
        </p>
        <p>
          <span>"Lorem ipsum dolot, set mat dolor, el mejor dise√±ador web del mundo"</span>
        </p>
      </div>
    </article>
  )
}

export default CustomerFeedback