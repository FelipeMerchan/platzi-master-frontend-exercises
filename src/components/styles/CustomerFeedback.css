.CustomerFeedback {
  color: #07080d;
}

.CustomerFeedback__title {
  margin-block-end: var(--space-2);
  font-family: var(--font-4);
  font-size: 40px;
  color: #c5345c;
  line-height: .7;
}

.CustomerFeedback__title>span {
  color: #66615f;
  padding-inline-start: var(--space-5);
}

.feedback p {
  font-size: 12px;
  margin-block-end: calc(var(--space) / 2);
}

.feedback:nth-of-type(1):after {
  content: '';
  display: inline-block;
  height: 1px;
  width: 80%;
  margin: var(--space-2) 0;
  background-color: #07080d;
}