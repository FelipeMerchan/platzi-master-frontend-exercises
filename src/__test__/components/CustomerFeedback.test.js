import React from 'react'
import { shallow, mount } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import CustomerFeedback from '../../components/CustomerFeedback'

describe('Probar el componente <CustomerFeedback />', () => {
  const customerFeedback = shallow(<CustomerFeedback />)
  test('Render del componente CustomerFeedback', () => {
    expect(customerFeedback.length).toEqual(1)
  })
  test('Render del título', () => {
    expect(customerFeedback.exists('.CustomerFeedback__title')).toBeTruthy()
  })
  test('Render la retroalimentación por parte de dos clientes', () => {
    expect(customerFeedback.children()).toHaveLength(3);
  });
})

describe('CustomerFeedback Snapshot', () => {
  const customerFeedback = create(<CustomerFeedback />)
  test('Comprobar la UI del componente CustomerFeedback', () => {
    expect(customerFeedback.toJSON()).toMatchSnapshot()
  })
})