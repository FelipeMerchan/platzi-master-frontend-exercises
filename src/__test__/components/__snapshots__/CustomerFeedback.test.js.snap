// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerFeedback Snapshot Comprobar la UI del componente CustomerFeedback 1`] = `
<article
  className="CustomerFeedback"
>
  <h2
    className="CustomerFeedback__title"
  >
    feedback
    <br />
     
    <span>
      de clientes
    </span>
  </h2>
  <div
    className="feedback"
  >
    <p>
      <strong>
        Sra. Lorem Ipsum
      </strong>
    </p>
    <p>
      <span>
        "Lorem ipsum dolot sit amet, muy buen trabajo!"
      </span>
    </p>
  </div>
  <div
    className="feedback"
  >
    <p>
      <strong>
        Sra. Lorem Ipsum
      </strong>
    </p>
    <p>
      <span>
        "Lorem ipsum dolot, set mat dolor, el mejor dise√±ador web del mundo"
      </span>
    </p>
  </div>
</article>
`;
